torch/optim/lr_scheduler.py

```diff
@@ -23,6 +22,42 @@ def __init__(self, optimizer, last_epoch=-1):
         self.step(last_epoch + 1)
         self.last_epoch = last_epoch
 
+        self.state = kwargs
+
+    def __getstate__(self):
+        return {
+            'state': self.state,
+            'last_epoch': self.last_epoch,
+        }
+
+    def __setstate__(self, state):
+        self.__dict__.update(state)
+
+    def state_dict(self):
+        """Returns the state of the scheduler as a :class:`dict`.
+
+        It contains two entries. One representing all variables passed through
+        kwargs and second representing the last_epoch:
+
+        * state - a dict holding current scheduler variables. Its content
+            differs between LR Scheduler classes.
+        * last_epoch - the last epoch of trying
+        """
+        state = {key: self.__dict__[key] for key in self.state.keys()}
+        return {'state': state, 'last_epoch': self.last_epoch}
+
+    def load_state_dict(self, state_dict):
+        """Loads the schedulers state.
+
+        Arguments:
+            state_dict (dict): scheduler state. Should be an object returned
+                from a call to :meth:`state_dict`.
+        """
+        state = state_dict['state']
+
+        self.last_epoch = state_dict['last_epoch']
+        self.__dict__.update(state)
```

Comment:
I think those two are pretty much equivalent to `__getstate__` and `__setstate__` in this implementation